### ISSUE TITLE
Tools for inspecting / debugging toots

### ISSUE THREAD

**Issue by alexjgriffith** [martianhi/mastodon.el#87]
## Suggestion
Here are some easy to use tools for seeing the properties of a toot. I've been using `mastodon-debug--inspect-toot` for inspecting / debugging toots. It pretty prints the results and dumps them into a new buffer. The focus shifts to the new buffer which is opened in another window.

I am not sure if this functionality has been implemented already, but I have not come across it. I am not sure if it should be included in the project itself. The functions can work stand alone when dropped in your init file.

``` elisp
(defun mastodon-debug--dump-json-in-buffer (name json)
  "Buffer NAME is opened and JSON in printed into it."
  (switch-to-buffer-other-window name)
  (progn (setf print-level nil
            print-length nil)
         (insert (pp json t))
         (goto-char 1)
         (emacs-lisp-mode)
         (message "success")))

(defun mastodon-debug--inspect-toot ()
  "Find next toot and dump its meta data into new buffer."
  (interactive)
  (mastodon-debug--dump-json-in-buffer
   (concat "*mastodon-debug-toot-"
           (int-to-string (mastodon-tl--property 'toot-id))
           "*")
  (mastodon-tl--property 'toot-json)))
```

I recommend binding `D` or `I` to  `mastodon-debug--inspect-toot` or its equivalent.

Currently that can be done by adding the following to your init file:
``` elisp
(define-key mastodon-mode-map (kbd "D") #'mastodon-debug--inspect-toot)
```

**NOTE:** Is this an appropriate place to make suggestions, or is their another place they should be submitted?

## Sample Screenshot
![screenshot 2017-04-26 02 22 46](https://cloud.githubusercontent.com/assets/6475397/25420703/5d2efe90-2a27-11e7-8916-bb9e2faa61a0.png)

**EDIT** added screen shot.

**Comment by jdenen** [comment]
> NOTE: Is this an appropriate place to make suggestions, or is their another place they should be submitted?

Here's fine.

> I am not sure if it should be included in the project itself.

I'd go ahead and submit a PR for them. I think it's fine to have developer functions in the project as long as they're not publicly facing. Meaning, let's not add them to the keymap or document them anywhere.

**Comment by jdenen** [comment]
Included in v0.6.1
